core/util.go

```diff
@@ -478,6 +478,36 @@ func VerifyCSR(csr *x509.CertificateRequest) error {
 	return errors.New("Unsupported CSR signing algorithm")
 }
 
+var (
+	oidExtensionRequest = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 14}
+	oidSubjectAltName   = asn1.ObjectIdentifier{2, 5, 29, 17}
+)
+
+// ExtensionsFromCSR extracts the set of requested extensions from a CSR
+func ExtensionsFromCSR(csr *x509.CertificateRequest) map[string][]byte {
+	extensionMap := map[string][]byte{}
+	for _, attr := range csr.Attributes {
+		if !attr.Type.Equal(oidExtensionRequest) || len(attr.Value) != 1 {
```

Comment:
Modulo the above comment, I think it makes sense to skip over things that are not relevant, but if we find a relevant thing and it has the wrong length, that should be an error.